### ISSUE TITLE
Potential improvements on `Parameters` subtypes

### ISSUE THREAD

**Issue by JordiBolibar** [ODINN-SciML/ODINN.jl#65]
Instead of having them as a stack of global parameters, gather them in a data structure (`struct` or `Dict`) in order to tidy things up.

**Comment by JordiBolibar** [comment]
An interesting alternative could be to use GeoParams.jl: https://github.com/JuliaGeodynamics/GeoParams.jl

**Comment by JordiBolibar** [comment]
Another easier alternative would be to use: https://github.com/mauro3/Parameters.jl

**Comment by mauro3** [comment]
Do note that `@kwdef` is now exported in Julia 1.9 (and thus officially supported), so maybe that is enough?  (can also be used in 1.8 via `Base.@kwdef`.

**Comment by JordiBolibar** [comment]
Yep, I've started the design of the new API and `@kwdef` is so far enough for me. Cool package!
